databricks_cli/permissions/cli.py

```diff
@@ -0,0 +1,188 @@
+               )
+@click.argument('object-type')
+@click.argument('object-id', required=False)
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def get_cli(api_client, object_type, object_id):
+    perms_api = PermissionsApi(api_client)
+    if not object_type:
+        click.echo('Possible object types are: {}'.format(PermissionTargets))
+    click.echo(pretty_format(perms_api.get_permissions(object_type, object_id)))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='List permission types')
+@click.argument('object_type')
+@click.argument('object_id')
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def list_permissions_types_cli(api_client, object_type, object_id):
+    perms_api = PermissionsApi(api_client)
+    click.echo(pretty_format(perms_api.get_possible_permissions(object_type, object_id)))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Add or modify permission types')
+@click.argument('object_type')
+@click.argument('object_id')
+@click.option('--group-name', metavar='<string>', cls=OneOfOption, default=None,
+              one_of=PERMISSIONS_OPTION)
+@click.option('--user-name', metavar='<string>', cls=OneOfOption, default=None,
+              one_of=PERMISSIONS_OPTION)
+@click.option('--service-name', metavar='<string>', cls=OneOfOption, default=None,
+              one_of=PERMISSIONS_OPTION)
+@click.option('--permission-level', metavar='<string>', default=None,
+              required=True)
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+# pylint:disable=unused-argument
+def add_cli(api_client, object_type, object_id, user_name, group_name, service_name,
+            permission_level):
+    perms_api = PermissionsApi(api_client)
+
+    if not user_name and not group_name and not service_name:
+        click.echo('Need --user-name, --service-name or --group-name')
+        return
+
+    if not permission_level:
+        click.echo('Need permission-level: {}'.format([e.value for e in PermissionLevel]))
+
+    # build the json for the use
+    perm_type = None
+    value = None
+    if user_name:
+        perm_type = PermissionType.user
+        value = user_name
+    elif group_name:
+        perm_type = PermissionType.group
+        value = group_name
+    elif service_name:
+        perm_type = PermissionType.service
+        value = service_name
+    else:
+        click.echo(
+            'Invalid permission-level must be one of {}'.format([e.value for e in PermissionLevel]))
+        return
+
+    permission = Permission(perm_type, value, Lookups.from_name(permission_level.upper()))
+    all_permissions = PermissionsObject()
+    all_permissions.add(permission)
+
+    if not all_permissions.to_dict():
+        click.echo('invalid permissions')
+        return
+
+    click.echo(pretty_format(perms_api.add_permissions(object_type, object_id, all_permissions)))
+
```

Comment:
We should add support for the following:
1. Set / update permissions
2. Ability to specify multiple permissions (not sure if this is possible already)

Reply:
I'll take note of that.

I also need to add some tests, and I'll cover those as part of the tests.